src/parser/FunctionParser.mts

```diff
@@ -27,7 +27,7 @@ type ConvertArrowParameterResult<T> =
   T extends keyof ArrowParameterConversions ? ArrowParameterConversions[T] : never;
 
 export abstract class FunctionParser extends IdentifierParser {
-  abstract parseStatementList(token: string | Token, directives?: string[]): ParseNode.StatementList;
+  abstract parseStatementList(token: string | Token, directives?: readonly string[]): ParseNode.StatementList;
```

Comment:
This can't be readonly as directives are written into it.

Reply:
To clarify, the implementation in `StatementParser` takes a writable array, not a `readonly` one. The fact this isn't an error is a bug in the compiler.

Reply:
I reverted this line.

Reply:
oh, when I change it to readonly the compiler doesn't give an error so I kept that.